qiskit_nature/second_q/operators/majorana_op.py

```diff
@@ -0,0 +1,521 @@
+# This code is part of a Qiskit project.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Majorana-particle Operator."""
+
+
+from __future__ import annotations
+
+import re
+from collections import defaultdict
+from collections.abc import Collection, Mapping
+from typing import Iterator, Sequence
+
+import numpy as np
+
+from qiskit_nature.exceptions import QiskitNatureError
+
+from .polynomial_tensor import PolynomialTensor
+from .sparse_label_op import _TCoeff, SparseLabelOp, _to_number
+from .fermionic_op import FermionicOp
+
+
+class MajoranaOp(SparseLabelOp):
+    r"""N-mode Majorana operator.
+
+    A ``MajoranaOp`` represents a weighted sum of Majorana fermion operator terms.
+    These terms are encoded as sparse labels, which are strings consisting of a space-separated list
+    of expressions. Each expression must look like :code:`_<index>`, where the :code:`<index>` is a
+    non-negative integer representing the index of the mode on which the Majorana
+    creation/annihilation operator is applied. The value of :code:`index` is bound by twice the
+    number of spin orbitals (``num_spin_orbitals``) of the operator (Note: since Python indices are
+    0-based, the maximum value an index can take is given by :code:`2 * num_spin_orbitals - 1`).
+
+    .. note::
+
+        For compatibility reasons (e.g. with mappers) :meth:`MajoranaOp.terms()` returns a list of
+        tuples of the form `("+", index)`, i.e. is compatible with the format for
```

Comment:
Yeah, I don't completely like it either. To be fair, it's not completely artificial, since any majorana operator is both a creator and annihilator, either + or - would be appropriate choices, but it's inconsistent with the "_index" labelling and hence confusing. I am not really sold on the "g" (or any other label) either though.
My main problem with the empty string in .terms() is the occurence in `QubitMapper.mode_based_mapping`. In line 260 of qubit_mapper.py there is a check for empty label for which I don't truly understand if it has any relevant use case. If this check can be safely omitted, then it's fine to use the empty string.
(On a side note, I feel that qubit_mapper needs an overhaul anyway, because some of the methods seem very specifically tailored to FermionicOp whereas this should be a base class which is applicable to any SparseLabelOp.)
```suggestion
        go back to tuples of the form `("", index)` compatible with the `_index` format and amend qubit_mapper accordingly when implementing the TernaryTreeMapper. (I would then assume that noting gets broken if all tests pass successfully.)
```

Reply:
Yes, I agree that the `QubitMapper` implementation needs to be improved. We did overhaul them a while back but made some questionable choices that time around.
The whole `mode_based_mapping` method should not be part of the base class and instead be a specific subclass (e.g. `ModeBaseMapper`) which can then be subclassed by those mappers which actually use it, without having this method be artificially present on _all_ mappers.

Another thing we wanted to change was also the use of `staticmethod` since this broke the initial implementation attempt of the [Bonsai mapper](https://journals.aps.org/prxquantum/abstract/10.1103/PRXQuantum.4.030314).

For now, since your PR does not include any mapper utilties, let's stick to the empty string label and not touch the `QubitMapper` stack. I will open a separate issue in a moment to track the changes that are required in that module.

Reply:
Back-referencing https://github.com/qiskit-community/qiskit-nature/issues/1289.